### PR TITLE
CMake CI [1/2]

### PR DESCRIPTION
**Pull request by sandro-elsweijer**
**_Describe your changes here:_**
Implements the foundation of the t8code cmake CI.
The testsuite file is the actually by github called workflow. It calls the other files afterwards.
The preparation step installs SC and p4est and uploads them as an artifact.
The sc and p4est file runs the sc and p4est tests.
The actual t8code tests are coming in the next PR

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: .github/workflows/tests_cmake_preparation.yml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with t8code; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+on:
+  workflow_call:
+    inputs:
+      MAKEFLAGS:
```
Comment (Davknapp) [inline .github/workflows/tests_cmake_preparation.yml:27]: Please add a description to the workflow inputs.

File: .github/workflows/tests_cmake_sc_p4est.yml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+#
+# Note: To manually enforce sc and p4est installation, either increase the counter
+#       in the "key:" entries of the sc and p4est steps or set the variables
+#       SC_IGNORE_CACHE and P4EST_IGNORE_CACHE to 1 in the respective steps.
+
+on:
+  workflow_call:
+    inputs:
+      MAKEFLAGS:
+        required: true
```
Comment (Davknapp) [inline .github/workflows/tests_cmake_sc_p4est.yml:37]: Please add a description to the inputs

**Review by Davknapp — CHANGES_REQUESTED** [review]

File: .github/workflows/tests_cmake_preparation.yml (hunk 1 above)
Comment (sandro-elsweijer) [reply .github/workflows/tests_cmake_preparation.yml:27]: done

File: .github/workflows/tests_cmake_sc_p4est.yml (hunk 2 above)
Comment (sandro-elsweijer) [reply .github/workflows/tests_cmake_sc_p4est.yml:37]: done

**Review by Davknapp — APPROVED** [review]

**Pull request merged**